sklearn/compose/tests/test_column_transformer.py

```diff
@@ -368,6 +368,37 @@ def test_column_transformer_sparse_stacking():
     assert_array_equal(X_trans[:, 1:], np.eye(X_trans.shape[0]))
 
 
+def test_column_transformer_mixed_cols_sparse():
+    pd = pytest.importorskip('pandas')
+    df = pd.DataFrame([['a', 1, True],
```

Comment:
I don't think we should require pandas because we need to test multiple scipy versions. Let's just use adtype='O' array

Reply:
That tests something different, though, right?
Can we have both tests?

Reply:
It should the same path I think (scipy.sparse.hstack getting an object array) 
But could indeed have both